### PR TITLE
dnsrobocert 3.24.1 (new formula)

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/dnsrobocert.rb [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+  resource "botocore" do
+    url "https://files.pythonhosted.org/packages/bf/9c/4b5d7f9197510a898215becf62fd9862be3fb0a2a63af5b656d24daf9de0/botocore-1.31.25.tar.gz"
+    sha256 "b8a40b0ca1e3c8290a4c0d473c8e1575d2e8b2ddc3c61dd8814c3976357cac84"
+  end
+
+  resource "certbot" do
+    url "https://files.pythonhosted.org/packages/0e/fd/192cb7d79f798432b1b2e7a9998b2f12dd589c50ff88e31553d95143b01c/certbot-2.6.0.tar.gz"
+    sha256 "c4de6bb0d092729650ed90a5bdb513932bdc47ec5f7f98049180ab8e4a835dab"
+  end
```
Comment (chenrui333) [inline Formula/dnsrobocert.rb:53]: `depends_on "certbot"`?

File: Formula/dnsrobocert.rb [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+  resource "distro" do
+    url "https://files.pythonhosted.org/packages/4b/89/eaa3a3587ebf8bed93e45aa79be8c2af77d50790d15b53f6dfc85b57f398/distro-1.8.0.tar.gz"
+    sha256 "02e111d1dc6a50abb8eed6bf31c3e48ed8b0830d1ea2a1b78c61765c2513fdd8"
+  end
+
+  resource "dns-lexicon" do
+    url "https://files.pythonhosted.org/packages/68/d0/0b67536c916324c37d4d53ecdc82dae8c683c2be7d4cac95820d812df127/dns_lexicon-3.14.0.tar.gz"
+    sha256 "506dc3a4e0f8a33bee2f71d3f24228f0c0e5dc167353d3ed7116e4d9206edc3e"
+  end
```
Comment (chenrui333) [inline Formula/dnsrobocert.rb:98]: `depends_on "lexicon"`?

**Comment by chenrui333** [comment]
with vendor out those two, we probably can get rid of some transitive dependencies.

**Comment by chenrui333** [comment]
need pth work for certbot.

**Review by chenrui333 — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]

**Pull request merged**